concept/thing/impl/AttributeImpl.java

```diff
@@ -88,7 +88,9 @@ public FunctionalIterator<ThingImpl> getOwners() {
 
     @Override
     public FunctionalIterator<ThingImpl> getOwners(ThingType ownerType) {
-        return ownerType.getSubtypes().map(ot -> ((ThingTypeImpl) ot).vertex).flatMap(
+        ThingType topLevelOwnerType = ownerType;
+        while (topLevelOwnerType.getSupertype().getOwns(getType().getValueType()).hasNext()) topLevelOwnerType = topLevelOwnerType.getSupertype();
+        return topLevelOwnerType.getSubtypes().map(ot -> ((ThingTypeImpl) ot).vertex).flatMap(
```

Comment:
Figure out which supertype owns this key attribute, then check all its subtypes.